onnxruntime/core/providers/webgpu/reduction/reduction_ops.cc

```diff
@@ -126,14 +162,28 @@ Status ReduceKernel<allow_multi_axes>::ComputeInternal(ComputeContext& context)
   if (input_axes.empty()) {
     if (noop_with_empty_axes_ || rank == 0) {
       // If axes is empty and noop_with_empty_axes_ is true, it is a no-op according to the spec
-      // If input tensor is a scalar, return the input tensor as is.
-      // This is not correct for ReduceLogSum and ReduceSumSquare
-      // TODO handle these cases separately.
-      auto output = context.Output(0, input_tensor->Shape());
-      if (output->DataRaw() != input_tensor->DataRaw()) {
-        ORT_RETURN_IF_ERROR(Info().GetDataTransferManager().CopyTensor(*input_tensor, *output));
+      // If input tensor is a scalar and it's not a ReduceLogSum or ReduceSumSquare, return the input tensor as is.
```

Comment:
ReduceL1 (absolute value) and ReduceL2 also need special handling if the input is a scalar.